packages/react-components/src/components/ConfigurationInitialStep.tsx

```diff
@@ -1,29 +1,63 @@
-import React, { ChangeEvent, FC } from 'react';
+import React, { FC, useEffect, useState } from 'react';
+import Button from './Button';
 import Label from './Label';
 import Tooltip from './Tooltip';
 import ErrorIcon from './icons/ErrorIcon';
-import Button from './Button';
 
+import { ConfigurationStep } from '../models/flare';
+import { fetchApiKeyValidation } from '../utils/setupConfiguration';
 import './ConfigurationGlobalStep.css';
 import './ConfigurationInitialStep.css';
+import { ToastKeys, toastManager } from './ToastManager';
 
 const ConfigurationInitialStep: FC<{
     show?: boolean;
+    configurationStep: ConfigurationStep;
     apiKey?: string;
-    errorMessage?: string;
-    isLoading?: boolean;
+    setApiKey: (apiKey: string) => void;
     onCancelConfigurationClick: () => void;
-    onSubmitApiKeyClick: () => void;
-    onApiKeyChange: (e: ChangeEvent) => void;
+    onApiKeyValidated: () => void;
 }> = ({
     show = false,
+    configurationStep,
     apiKey = '',
-    errorMessage = '',
-    isLoading = false,
+    setApiKey,
     onCancelConfigurationClick,
-    onSubmitApiKeyClick,
-    onApiKeyChange,
+    onApiKeyValidated,
 }) => {
+    const [errorMessage, setErrorMessage] = useState('');
+    const [isLoading, setIsLoading] = useState(false);
+    const handleApiKeyChange = (e): void => setApiKey(e.target.value);
+
+    const handleSubmitApiKey = (): void => {
+        setIsLoading(true);
+        fetchApiKeyValidation(apiKey)
+            .then(() => {
+                setErrorMessage('');
+                setIsLoading(false);
+                onApiKeyValidated();
+            })
+            .catch((error: any) => {
+                setErrorMessage(error.data);
+                setIsLoading(false);
+                toastManager.show({
+                    id: ToastKeys.ERROR,
+                    isError: true,
+                    content: 'Something went wrong. Please review your form.',
+                });
+            });
+    };
+
+    const isFormValid = (): boolean => {
+        return apiKey.length > 0;
```

Comment:
We don't have to do this, but we could do a couple more checks on the apiKey here:

1. Can ensure that the apiKey starts with `fw_`
2. Can ensure that the total length is 44 chars, including the `fw_`

I'm reasonably certain we don't expect that to change ever.

Reply:
Just in case it does change in the future, we already send a call to validate the API key, so I feel like it isn't that 
useful to add UI validation on top

Reply:
> I'm reasonably certain we don't expect that to change ever.

I wouldn't validate the firework API key. We could definitely change the format in the future.